deku-p/src/core/block_storage/tests/block_storage_tests.ml

```diff
@@ -0,0 +1,68 @@
+open Deku_stdlib
+open Deku_crypto
+open Deku_consensus
+open Deku_concepts
+
+let block_testable = Alcotest.testable Block.pp Block.equal
+
+let identity =
+  let secret = Ed25519.Secret.generate () in
+  let secret = Secret.Ed25519 secret in
+  Identity.make secret
+
+let block ~default_block_size =
+  let above = Genesis.block in
+  let withdrawal_handles_hash = BLAKE2b.hash "potato" in
+  let producer = Producer.empty in
+  Producer.produce ~identity ~default_block_size ~above ~withdrawal_handles_hash
+    producer
+
+let file_hash =
+  let randn = Stdlib.Random.int 230 in
+  Deku_crypto.BLAKE2b.hash (Int.to_string randn) |> BLAKE2b.to_hex
+
+let uri = Uri.of_string (Format.sprintf "sqlite3:/tmp/%s.db" file_hash)
```

Comment:
Can we have in memory database?

Reply:
Not sure. It looks like this is an option in [SQLite itself](https://www.sqlite.org/inmemorydb.html), but probably won't be available in the OCaml ecosystem.